### DISCUSSION TITLE
How to create dependent optionmenu/combobox

### DISCUSSION THREAD

**Discussion by Rahuldt1997** [TomSchimansky/CustomTkinter#1903] [Q&A]
How to create dependent optionmenu/combobox, where list 2's items depend on list 1's selected value and same for list 3(depend on list 2's item). i have created the 3 combobox in normal tkinter, and it is working perfectly but in customtk it is only showing the 1st combobox values, I am using customtkinter.StringVar for that. Dose customtkinter.StringVar support .trace method ?

class Ncase:
    x=state_city_court
    
    def __init__ (self,master): 
        self.master = master
        #
        #
        
        # Create Variables for combo box
        self.state_var = customtkinter.StringVar(master)
        self.state_var.set('Select State')
        self.state_var.trace('w', self.on_state_change)

        self.city_var = customtkinter.StringVar(master)
        self.city_var.set('Select City')
        self.city_var.trace('w', self.on_city_change)

        self.court_var = customtkinter.StringVar(master)
        self.court_var.set('Select Court')
    
        #
        #

        #state
        lstate=customtkinter.CTkLabel(master,text="State",padx=5,pady=5,anchor="w",justify="left",font=("arial",16))
        lstate.grid(column=0,row=2,sticky=E,pady=(0,20))
        state_combo = customtkinter.CTkOptionMenu(self.master, variable=self.state_var, values=list(self.x.keys()), state='readonly',width=300,font=("arial",16))
        state_combo.grid(column=1,row=2,padx=5,pady=(0,20))

        #city
        lcity=customtkinter.CTkLabel(master,text="City",padx=5,pady=5,anchor="w",justify="left",font=("arial",16))
        lcity.grid(column=2,row=2,sticky=E,pady=(0,20))
        self.city_combo = customtkinter.CTkOptionMenu(self.master, variable=self.city_var, values=['Select City'], state='readonly',width=300,font=("arial",16))
        self.city_combo.grid(column=3,row=2,padx=5,pady=(0,20))    

        #name of court
        cname=customtkinter.CTkLabel(master,text="Select Court",padx=5,pady=5,anchor="w",justify="left",font=("arial",16))
        cname.grid(column=4,row=2,sticky=E,pady=(0,20))
        self.court_combo = customtkinter.CTkOptionMenu(self.master, variable=self.court_var, values=['Select Court'], state='readonly',width=300,font=("arial",16))
        self.court_combo.grid(column=5,row=2,padx=5,pady=(0,20))
        
        
        #
        #
    
            
    def on_state_change(self, *args):
        selected_state = self.state_var.get()
        self.city_combo['values'] = list(self.x.get(selected_state, {}).keys())
        self.city_var.set('Select City')
        self.court_var.set('Select Court')

    def on_city_change(self, *args):
        selected_state = self.state_var.get()
        selected_city = self.city_var.get()
        
        self.court_combo['values'] = self.x.get(selected_state, {}).get(selected_city, [])
        print("adswad")
        self.court_var.set('Select Court')

**Comment by avalon60** [comment]
You don't have any callback functions defined. You need to use the `command` option, to get the widgets to interact.

```
   state_combo = customtkinter.CTkOptionMenu(self.master, variable=self.state_var, values=list(self.x.keys()),
                                              state='readonly', width=300, font=("arial", 16), command=on_state_change)
...
  def on_state_change(self):
     # Here place code to configure any cascaded properties for the "cities" widget. E.g. Set appropriate values (list of cities for selected state) and enable the widget. 
```     
 You seem to have started the last part, but you need to maybe start off with disabled widgets and enable them, as the user moves down, to ensure data integrity between widget values lists. 

**Also you appear to be using Tkinter specific syntax for configuring values. You need to use self.city_combo.configure(values=...).**
